### PR TITLE
[#10981] Add team members as an option to generate values

### LINKED ISSUE #10981: Instructor setting question options: support generating options from team members
Shouldn't there be options `team members` and `team members excluding self` here?
image

### PR DESCRIPTION
**Pull request by daongochieu2810**
Fixes #10981

**Outline of Solution**

- FE:
  - Add `team members` and `team members (excluding self)` into the dropdown selection
- BE:
  - Return team member names if the above options are selected

### CODE REVIEW

File: src/main/java/teammates/logic/core/FeedbackQuestionsLogic.java [hunk 1]
```diff
@@ -627,6 +627,22 @@ public void populateFieldsToGenerateInQuestion(FeedbackQuestionAttributes feedba
                 Assumption.fail("Course disappeared");
             }
             break;
+        case OWN_TEAM_MEMBERS_INCLUDING_SELF:
+            //fallthrough
```
Comment (madanalogy) [inline src/main/java/teammates/logic/core/FeedbackQuestionsLogic.java:631]: Comment is unnecessary

**Review by madanalogy — CHANGES_REQUESTED** [review]
Seems like a simple addition. Could you record a quick gif or something of the sort to demonstrate that it works as intended?

**Comment by daongochieu2810** [comment]
@madanalogy 
ezgif com-gif-maker (2)

**Comment by madanalogy** [comment]
@daongochieu2810 remove the unnecessary comment and add tests to `FeedbackQuestionsLogicTest` for this addition.

**Comment by damithc** [comment]
While the change required may be small, this is a non-trivial feature addition. Need to test thoroughly. Most likely test cases need to be updated.

File: src/main/java/teammates/logic/core/FeedbackQuestionsLogic.java [hunk 2]
```diff
@@ -593,7 +593,6 @@ public void populateFieldsToGenerateInQuestion(FeedbackQuestionAttributes feedba
         case NONE:
             break;
         case STUDENTS:
-            //fallthrough
```
Comment (moziliar) [inline src/main/java/teammates/logic/core/FeedbackQuestionsLogic.java:596]: Not sure if removing a `//fallthrough` comment in Java is a good idea in the case of fallthrough; it does has its own semantic contribution.

File: src/main/java/teammates/logic/core/FeedbackQuestionsLogic.java [hunk 3]
```diff
@@ -627,6 +625,23 @@ public void populateFieldsToGenerateInQuestion(FeedbackQuestionAttributes feedba
...
+        case OWN_TEAM_MEMBERS:
+            if (teamOfEntityDoingQuestion != null) {
+                List<StudentAttributes> teamMembers = studentsLogic.getStudentsForTeam(teamOfEntityDoingQuestion,
+                        feedbackQuestionAttributes.getCourseId());
+
+                if (generateOptionsFor == FeedbackParticipantType.OWN_TEAM_MEMBERS) {
+                    teamMembers.removeIf(teamMember -> teamMember.getEmail().equals(emailOfEntityDoingQuestion));
+                }
+
+                teamMembers.forEach(teamMember -> {
```
Comment (moziliar) [inline src/main/java/teammates/logic/core/FeedbackQuestionsLogic.java:638]: Inline the lambda?

**Review by hcwong — APPROVED** [review]
changes look fine :)

**Review by madanalogy — APPROVED** [review]

File: src/main/java/teammates/logic/core/FeedbackQuestionsLogic.java (hunk 3 above)
Comment (madanalogy) [reply src/main/java/teammates/logic/core/FeedbackQuestionsLogic.java:638]: @daongochieu2810 do you intend to adopt this suggestion? Would be good to address comments to give reviewers some resolution

Comment (daongochieu2810) [reply src/main/java/teammates/logic/core/FeedbackQuestionsLogic.java:638]: sorry I forgot. Yes I will make the changes

**Pull request merged**